components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/listener/WorkflowListener.java

```diff
@@ -270,17 +270,44 @@ void doPostAddAssociation(String associationName, String workflowId, String even
      * Trigger before listing workflows of a tenant
      *
      * @param tenantId Tenant ID
+     * @param limit Limit
+     * @param offset Offset
+     * @param filter Filter
      * @throws WorkflowException
      */
+    void doPreListPaginatedWorkflows(int tenantId, int limit, int offset, String filter) throws WorkflowException;
```

Comment:
As these are new methods to interface, these needs to be default methods, (same for other new methods)

The implementation here should call `doPreListPaginatedWorkflows(tenantId)`  internally

Reply:
Since old methods have been annotated as deprecated, Do we need to call those deprecated methods inside newly added methods?